### DISCUSSION TITLE
How can I build static library in Xcode for macOS?

### DISCUSSION THREAD

**Discussion by lydasia** [libusb/libusb#1481] [Q&A] [answered]
Hi,

The provided project located in libusb-1.0.27/Xcode is for dynamic library.
I got .dylib file after build without any issue. 
Is there any guide for modifying the project for a static library?

Thanks a lot!

**Comment by seanm** [comment]
Probably because libusb is LGPL, static linking is discouraged.

**Reply by lydasia** [reply]
Thanks for your reply, I only plan to use static linking in fully open source software.

**Comment by Youw** [comment, marked as answer]
If you plan to use XCode project specifically, you can poke around in XCode project settings locally to change it.
There is no external option to change it. (Some projects even have (technically) different XCode projects for different build variants, like shared/static).

If XCode project is not a requirement for you - use Autotools to build the project and static library may be selected during configuration (check CI scripts if you need an example how to do so).

The other alternative is to use [libusb-cmake](https://github.com/libusb/libusb-cmake). It even allows you to generate an XCode project for static variant of the library build.
NOTE: that project is not portable/movable, i.e. you need to generate it each time you need to copy/move the project around on the filesystem/among different machines.

**Reply by lydasia** [reply]
Thank you very much for your answer!
